Source/player.h

```diff
@@ -243,9 +252,10 @@ struct Player {
 	float progressToNextGameTickWhenPreviewWasSet;
 	int _plid;
 	int _pvid;
-	spell_id _pSpell;
-	spell_type _pSplType;
-	int8_t _pSplFrom; // TODO Create enum
+	/* @brief next queued spell */
+	SpellCastInfo queuedSpell;
+	/* @brief the spell that is currently casted */
+	SpellCastInfo castedSpell;

```

Comment:
```suggestion
	SpellCastInfo currentSpell;
```

Reply:
Oh, I might need to read more of the code to understand it better :D

How about "inflightSpell"?

Reply:
In some way the queued spell is also in-flight or? 🤔 

The pr tries to distinguish between the spell (what spell, level, etc.) that is currently performed by the player (`_pmode` = `PM_SPELL`) and what is the next spell (`destAction` = `ACTION_SPELL`/`ACTION_SPELLWALL`/`ACTION_SPELLMON`/`ACTION_SPELLPLR`).
In master/vanilla for both the same variables were used and this introduced the double casting bug.

Reply:
no, it's pending on the runway

Reply:
Renamed to `executedSpell`.
Hope this is better. 🙂

Reply:
yeah that works :)